list.py

```diff
@@ -227,17 +227,21 @@ def remove(self, index):
 
 
         if index == 0:
+            elem = self.head.elem
             self.head = self.head.next
             self._length -= 1
-            return
+            return elem
 
         prev_node = self.find_node(index - 1)
+        elem = prev_node.next.elem
 
         if index == self._length - 1:
             self.tail = prev_node
         
         prev_node.next = prev_node.next.next
         self._length -= 1
+        
+        return elem
```

Comment:
I looked at the LL documentation and the [`remove()`](https://doc.rust-lang.org/src/alloc/collections/linked_list.rs.html#918) implementation in Rust, where this method returns the deleted value. The same applies to [`pop()`](https://docs.rs/queues/latest/src/queues/lib.rs.html#284) in Queue.